docker/tinode/entrypoint.sh

```diff
@@ -89,42 +89,32 @@ if [ ! -z "$IOS_UNIV_LINKS_APP_ID" ] ; then
 EOM
 fi
 
-run_init_db=false
-run_tinode=false
-case "$SERVICES_TO_RUN" in
-"init-db")
-  run_init_db=true
-  ;;
-"tinode")
-  run_tinode=true
-  ;;
-"both")
-  run_init_db=true
-  run_tinode=true
-  ;;
-*)
-  echo "Invalid val for SERVICES_TO_RUN env var. Can be either 'init-db' or 'tinode' or 'both'."
-  exit 1
-  ;;
-esac
-
-echo "Will run init-db: ${run_init_db}, tinode: ${run_tinode}"
-
-touch /botdata/tino-password
-
-if [ "$run_init_db" == "true" ]; then
-  init_args=("--reset=${RESET_DB}" "--upgrade=${UPGRADE_DB}" "--config=${CONFIG}")
-  # Maybe load sample data?
-  if [ ! -z "$SAMPLE_DATA" ] ; then
-    init_args+=("--data=$SAMPLE_DATA")
-  fi
-  # Initialize the database if it has not been initialized yet or if data reset/upgrade has been requested.
-  ./init-db "${init_args[@]}" | grep "usr;tino;" > /botdata/tino-password
+init_args=("--reset=${RESET_DB}" "--upgrade=${UPGRADE_DB}" "--config=${CONFIG}")
+# Maybe load sample data?
+if [ ! -z "$SAMPLE_DATA" ] ; then
+	init_args+=("--data=$SAMPLE_DATA")
+fi
+init_stdout=./init-db-stdout.txt
+init_stderr=./init-db-stderr.txt
+# Initialize the database if it has not been initialized yet or if data reset/upgrade has been requested.
+./init-db "${init_args[@]}" 1>$init_stdout 2>$init_stderr
+init_result=$?
+cat $init_stderr >&2
+if [ $init_result != 0 ]; then
+	echo "./init-db failed. Quitting."
+	exit 1
+fi
+
+# If sample data was provided, try to find Tino password.
+if [ ! -z "$SAMPLE_DATA" ] ; then
+	grep "usr;tino;" $init_stdout > /botdata/tino-password
 fi
 
-if [ "$run_tinode" != "true" ]; then
-  # If we don't want to run tinode, we are done.
-  exit 0
+# Check if the init-db output contains the magic string.
+grep -q "All done" $init_stderr
```

Comment:
Please consider checking for exit code:
```
if [ $? -eq 0 ]
then
  echo "Success"
else
  echo "Failure" >&2
fi
```

Reply:
Done below.

Reply:
Then you don't need the `grep -q "All done" $init_stderr` line anymore.